### DISCUSSION TITLE
Change chargin power [Wallbox Pulsar Plus]

### DISCUSSION THREAD

**Discussion by gmelfedo** [lbbrhzn/ocpp#309] [Q&A] [answered]
Hi everybody!
There is a way to change the charging power before or during a session?
![image](https://user-images.githubusercontent.com/22681200/149840894-2a42a495-8b6c-45f3-a780-88295a07cb46.png)

**Comment by GrantKeymer** [comment, marked as answer]
Hi @gmelfedo - Welcome!

I'm also using a Wallbox Pulsar Plus with v0.4.6 of the OCPP integration.  My experience is the same as yours ... Maximum_Current control is Unavailable.  To change the charging current, I need to use My Wallbox Portal or My Wallbox App.

However, the owner of this repo (lbbrhzn) has the Maximum_Current control working, using a development version of the code.

So it is possible for this control to work, but we need to be patient for a while longer, and may need to help with debugging, according to server logs etc.

Anyway, it's great to have you here in this forum, another Wallbox Pulsar Plus user.  My plan is to use this OCPP integration to automate charging from solar power.  At present, I need to keep looking at the sky in case a cloud comes over and then I end up using grid power.  I also need to adjust charging power according to time of day and time of year.

With a properly constructed HA automation and an integration connecting to my solar power system, it should be possible for all these adjustments to be made by themselves, giving me a fully-charged EV, using only solar power.  I already tried the Energy Meter option from Wallbox and couldn't make it work.  But I can see now that the Wallbox Eco-Smart upgrade is not actually very smart.  Using HA will allow me to develop something much smarter.

**Comment by gmelfedo** [comment]
Thank you so much for the fast response!
I'm planning to automatic balance the available power too.
Can't wait to have it working so I can brick something else in my house while playing on Home Hassistant!

**Reply by GrantKeymer** [reply]
Well hopefully you won't brick anything!

We can compare experiences and help each other along the way.

**Comment by lbbrhzn** [comment]
@gmelfedo, we have added a slider to change the max charge current. It is only supported when the charger supports the SmartCharging profile. The UI makes the slider unavailable when the charger is offline or when the charger does not support that profile. Unfortunately, it appears that there are some issues making the slider available again when the charger re-connects and/or when the value of the slider is restored after reboot.
